### ISSUE TITLE
Code files should all have .f90 (or .F90) extension instead of .f

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/NCEPLIBS-g2#151]
All the code files have a .f extension. Yet they use F90 features.

The compilers are mostly OK with this because they just ignore the extension. But it is confusing and also confuses doxygen.

The simplest answer would be to rename each file to a .f90 extension.

@Hang-Lei-NOAA @BrianCurtis-NOAA @kgerheiser any objections to this?

**Comment by Hang-Lei-NOAA** [comment]
There are some codes using F77 standard

**Comment by edwardhartnett** [comment]
F77 is valid F90, but F90 is not valid F77.

So we can rename all the files to .F90, or leave the F77 ones .f.

**Comment by kgerheiser** [comment]
It's not a matter of whether they use Fortran 90/77 features. It's fixed formatting vs. free-formatting. You couldn't just re-name them without changing the formatting or adding a compiler option.

**Comment by edwardhartnett** [comment]
The only thing I seem to have to change are the line continuations.

Fixed format likes the continuation line on the second line, like this:

```
      integer :: mappds(ipdstmplen), intbmap(ngrdpts),
     $     mapdrs(idrstmplen)

```
Whereas F90 likes it at the end of the first line:

```
      integer :: mappds(ipdstmplen), intbmap(ngrdpts), &
           mapdrs(idrstmplen)

```

Once I changed all the continuation lines, addfield.f becomes addfield.F90, and compiles and passes all tests.

I'll put up a PR with just that one file change. If we all agree it looks good, I will change the continuation characters in the rest of the codebase and we will convert everything to free-format F90 files.

Perhaps we can convert all our F77 code before the 50th anniversary of F77!
